jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jsonb/JsonbDiagnosticsCollector.java

```diff
@@ -124,4 +138,36 @@ private void addJsonbTransientSpecificDiagnostics(Diagnostic d) {
         d.setMessage(JsonbConstants.ERROR_MESSAGE_JSONB_TRANSIENT);
         d.setCode(JsonbConstants.DIAGNOSTIC_CODE_ANNOTATION_TRANSIENT_FIELD);
     }
+    
+    private boolean hasJsonbTransient(IAnnotatable annotable) throws JavaModelException {
+        for (IAnnotation annotation : annotable.getAnnotations()) {
+            String annotationName = annotation.getElementName();
+            if (annotationName.equals(JsonbConstants.JSONB_TRANSIENT))
+                return true;
+        }
+        return false;
+    }
+    
+    private boolean hasOtherJsonbAnnotation(IAnnotatable annotable) throws JavaModelException {
+        for (IAnnotation annotation : annotable.getAnnotations()) {
+            String annotationName = annotation.getElementName();
+            if (annotationName.startsWith(JsonbConstants.JSONB_PREFIX) 
+                    && !annotationName.equals(JsonbConstants.JSONB_TRANSIENT)) {
+                return true;
+            }
+        }
+        return false;
+    }
+    
+    private boolean isSetterOf(IMethod method, IField field) {
```

Comment:
I think these two methods can be combined into one method called `isAccessor(GET, method, field)`, where GET is the constant "get", and similarly for SET. This will avoid duplication as we can do in the method body simply `ACCESSOR_TYPE + fieldName.substring(0, 1).toUpperCase() + fieldName.substring(1);`

Reply:
Fixed